src/Http/Controllers/Admin/ForgotPasswordController.php

```diff
@@ -20,23 +23,45 @@ class ForgotPasswordController extends Controller
 
     use SendsPasswordResetEmails;
 
-    public function broker()
-    {
-        return Password::broker('twill_users');
-    }
+    /**
+     * @var PasswordBrokerManager
+     */
+    protected $passwordBrokerManager;
 
-    public function showLinkRequestForm()
-    {
-        return view('twill::auth.passwords.email');
-    }
+    /**
+     * @var ViewFactory
+     */
+    protected $viewFactory;
 
     /**
      * Create a new controller instance.
      *
+     * @param Application $app
+     * @param PasswordBrokerManager $passwordBrokerManager
+     * @param ViewFactory $viewFactory
+     * @param Config $config
      * @return void
      */
-    public function __construct()
-    {
+    public function __construct(
+        Application $app,
+        PasswordBrokerManager $passwordBrokerManager,
+        ViewFactory $viewFactory,
+        Config $config
+    ) {
+        parent::__construct($app, $config);
+
+        $this->passwordBrokerManager = $passwordBrokerManager;
+        $this->viewFactory = $viewFactory;
         $this->middleware('twill_guest');
     }
+
+    public function broker()
+    {
+        return $this->passwordBrokerManager->broker('twill_users');
```

Comment:
`PasswordBrokerManager` can be injected in this method instead of the constructor.

Reply:
This is actually not true because this method is called in vendor/laravel/framework/src/Illuminate/Foundation/Auth/SendsPasswordResetEmails.php:33

```
        $response = $this->broker()->sendResetLink(
            $request->only('email')
        );
```
so there are no function arguments.

Reply:
Yes, you are right.

Reply:
Can I resolve this one?

Reply:
Yes.